docs/design/gluster-block-provisioning.md

```diff
@@ -0,0 +1,350 @@
+# Gluster-Block Provisioning End-To-End Design
+
+## Motivation
+
+While GlusterFS volumes can be used for RWO as well as RWX (and ROX) volumes,
+for some workloads, specifically metadata-heavy, the RWO support is not
+satisfactory with respect to performance. Furthermore, the scalability
+in number of volumes is currently not very good.
+
```

Comment:
Ok, we sorted it out. The challenges with RWO are three-fold:

1. RWO volumes should have protection from concurrent mounting. (Locking to a host.) This is currently not enforced for GlusterFS but could be achieved by modifying the GlusterFS mount plugin.
2. The performance of the GlusterFS file volumes does not match the expectations of many users who want to run dedicated workloads like databases and metrics on the RWO volumes.
    This is a fundamental problem in Gluster, because specifically metadata operations are slow due to the distributed/shared nature.
3. The scalability in the number of volumes that can be hosted is limited for GlusterFS volumes. But because the RWO volumes can't be shared, specifically for RWO we need a very good scalability.

I will clarify the text a bit in those lines.